Add article about integration of PG and ClickHouse (#1271)
aiven/devportal: docs/products/clickhouse/howto/integrate-pg.rst

Integrate with PostgreSQL®
=============================

You can integrate Aiven for ClickHouse® with either *Aiven for PostgreSQL* service located in the same project, or *an external PostgreSQL endpoint*.

Behind the scenes the integration between Aiven for ClickHouse and PostgreSQL relies on `ClickHouse PostgreSQL Engine <https://clickhouse.com/docs/en/engines/table-engines/integrations/postgresql>`_.

.. note::

    Aiven for ClickHouse service integrations are available for Startup plans and higher.

Prerequisites
-------------

> You will need:

Reviewer: I would always reaffirm what they're doing here 
```suggestion
To connect Aiven for Clickhouse to PostgreSQL you will need:
```

* Aiven for ClickHouse service
* Aiven for PostgreSQL service or a self-hosted PostgreSQL service

> * If you use the self-hosted PostgreSQL service, an external PostgreSQL endpoint should be configured in **Service Integrations**

Reviewer: I put this as a Tip rather than an extra dot in the Kafka version

Create an integration
----------------------

You can create an integration with the help of :ref:`Aiven CLI <avn_service_integration_create>`, or through the `Aiven console <https://console.aiven.io/>`_ by following these steps:

1. Log in to the `Aiven web console <https://console.aiven.io/>`_, choose the right project, and select your Aiven for ClickHouse service.
#. In the *Overview* tab, scroll till the section *Service integrations* and click on **Set up integration**. A modal window with available integration options will appear.
#. Select PostgreSQL from the list of the options.
#. Choose your service from the list. This is also a place where you can trigger creation of a new Aiven for PostgreSQL service.
#. Click on **Enable**.
#. Your PostgreSQL service will be added to the list of enabled service integrations.
#. You can now close *Service integrations* modal window.
#. The PostgreSQL integration database will now be added.

The newly created database name has the following format: ``service_PG_SERVICE_NAME_PG_DATABASE_PG_SCHEMA``, for example, ``service_myPGService_myDatabase_mySchema``.

.. note::

>     When connecting to an Aiven for PostgreSQL, we connect as the main service user of that service, which has access to all the PostgreSQL tables. SELECT and INSERT privileges are granted to the main service user. It is up to the main service user to grant access to other users.

Reviewer: should we state here that is `avnadmin`?

Reviewer: also, what is this meaning "It is up to the main service user to grant access to other users"?

Update PostgreSQL integration settings
-----------------------------------------

> When connecting to a PostgreSQL service, ClickHouse needs to know the name of the PostgreSQL schema and database you want to access. By default these settings are set to the public schema in the ``defaultdb``. However, you can update these values by following next steps.

Reviewer: ```suggestion
When connecting to a PostgreSQL service, ClickHouse needs to know the name of the PostgreSQL schema and database you want to access. By default these settings are set to the ``public`` schema in the ``defaultdb``. However, you can update these values by following next steps.
```

.. note::

    Currently the configurations can be set only with the help of CLI command :ref:`avn service integration-update <avn service integration-update>`.

1. Get *the service integration id* by requesting the full list of integrations. Replace ``CLICKHOUSE_SERVICE_NAME`` and ``PG_SERVICE_NAME`` with the names of your services:

.. code::

    avn service integration-list CLICKHOUSE_SERVICE_NAME | grep PG_SERVICE_NAME

> 2. Update the configuration settings using the service integration id retrieved in the previous step and your integration settings. Replace ``SERVICE_INTEGRATION_ID``, ``PG_DATABASE`` and ``PG_SCHEMA`` with your values:

Reviewer: Is this meaning that only one schema can be attached between CH and PG? What if I need to attach multiple schemas?

Author: databases is an array, so it accepts multiple items. 
Or, you mean more than one schema per same database? This should work - but this will create two different databases in CH, and not combining schemas. Will it make sense from PG perspective?
`"databases":[{"database":"PG_DATABASE","schema":"PG_SCHEMA1"}, {"database":"PG_DATABASE","schema":"PG_SCHEMA2"},]`

Author: I corrected the description to mention that you can have more than one

Reviewer: Then it's fine, since we basically create the integration with the avnadmin user.
I believe, for external DBS, they'll likely create an integration per user so it should be fine too

.. code::

    avn service integration-update SERVICE_INTEGRATION_ID \
    --user-config-json '{
        "databases":[{"database":"PG_DATABASE","schema":"PG_SCHEMA"}]
    }'

Read and store data
-------------------
In Aiven for ClickHouse you can read data by running SELECT command. Replace ``PG_SERVICE_NAME``, ``PG_DATABASE``, ``PG_SCHEMA`` and ``PG_TABLE`` with your values and run:

.. code:: sql

    SELECT * FROM service_PG_SERVICE_NAME_PG_DATABASE_PG_SCHEMA.PG_TABLE

.. note::

>     ClickHouse is strict about allowed symbols in database and table names. You can use backticks around the names when running ClickHouse requests, particularly in the cases when the name contains dashes.

Reviewer: Would be cool to show an example here

Write data to PostgreSQL table
-------------------------------

> You can also bring the entries from ClickHouse table into the PostgreSQL table. Replace ``PG_SERVICE_NAME``, ``PG_DATABASE``, ``PG_SCHEMA`` and ``PG_TABLE`` with your values:

Reviewer: ```suggestion
You can also insert rows from the ClickHouse table into the PostgreSQL table. Replace ``PG_SERVICE_NAME``, ``PG_DATABASE``, ``PG_SCHEMA`` and ``PG_TABLE`` with your values:
```
